properties/basic.py

```diff
@@ -4,17 +4,35 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
+import collections
 import datetime
 import uuid
 
 import numpy as np
 from six import integer_types
 from six import string_types
+from six import with_metaclass
 
 from .utils import undefined
 
+PropertyTerms = collections.namedtuple(
+    'PropertyTerms',
+    ('name', 'cls', 'args', 'kwargs'),
+)
+
+
+class ArgumentWrangler(type):
+    """Stores arguments to property initialization for later use"""
+
+    def __call__(cls, *args, **kwargs):
+        """Wrap __init__ call in GettableProperty subclasses"""
+        instance = super(ArgumentWrangler, cls).__call__(*args, **kwargs)
+        instance._args = args
```

Comment:
It doesn't know the name at that point (this is before binding to the `HasProperties` instance, so `PropertiesMetaclass.__new__` hasn't been called yet). Otherwise we could do that. The name is not totally necessary.